Assets/PinAnimator.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PinAnimator : MonoBehaviour
+{
+    // Start is called before the first frame update
+    [SerializeField]
+    private float maxDist;
+
+    private float time;
+
+    [SerializeField]
+    private float delay = 0f;
+
+    [SerializeField]
+    private AnimationCurve easeCurve;
+    void Start()
+    {
+        var projectileController = GetComponentInParent<GunController>();
+
+        projectileController.onFire += (GunStats stats) => recoil();
```

Comment:
Please let Recoil() take GunStats as argument so we can += and -= the method instead

Reply:
Resolved this myself. Made this class actually use the BarrelAnimator that I already wrote.